### PR TITLE
TST/REF: arithmetic tests for BooleanArray + consolidate with integer masked tests

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Follow-up on https://github.com/pandas-dev/pandas/pull/34454

- Move a bunch of tests from `integer/test_arithmetic.py` to `masked/test_arithmetic.py` so those can be shared with boolean (and once merged I can add float as well in the FloatingArray PR)
- Added some more arithmetic tests to `boolean/test_arithmetic.py` to follow the same pattern as we have in `integer/test_arithmetic.py`. 
- The commong integer/boolean tests uncovered a bug in the BooleanArray implementation when dealing with a `pd.NA` scalar as other operand.

### CODE REVIEW

File: pandas/tests/arrays/boolean/test_arithmetic.py [hunk 1]
```diff
@@ -13,30 +15,87 @@ def data():
...
-        ops = getattr(s, op)
-        opa = getattr(data, op)
 
-        # invalid scalars
-        with pytest.raises(TypeError):
-            ops("foo")
-        with pytest.raises(TypeError):
-            ops(pd.Timestamp("20180101"))
+@pytest.fixture
+def b():
```
Comment (jbrockmendel) [inline pandas/tests/arrays/boolean/test_arithmetic.py:24]: would it be burdensome to give a and b more informative names?

**Comment by jbrockmendel** [comment]
small comment; LGTM

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/boolean/test_arithmetic.py:24]: Do you have a suggestion what you would find clearer?

Basically, before moving it to a fixture, I had:

```
def test_..():
    a = pd.array([True] * 3 + [False] * 3 + [None] * 3, dtype="boolean")
    b = pd.array([True, False, None] * 3, dtype="boolean")

    result = op(a, b)
```

in each test, and I would think in such a context you would be fine with those names? 
And although I like defining those variables inside the test as more explicit, it was also a bit repetitive, so I moved it to a fixture.

File: pandas/_testing.py [hunk 2]
```diff
@@ -2758,3 +2759,28 @@ def get_cython_table_params(ndframe, func_names_and_expected):
             if name == func_name
         ]
     return results
+
+
+def get_op_from_name(op_name):
```
Comment (jreback) [inline pandas/_testing.py:2764]: can you type op_name and return type

File: pandas/core/arrays/boolean.py [hunk 3]
```diff
@@ -717,11 +717,20 @@ def boolean_arithmetic_method(self, other):
...
             if mask is None:
                 mask = self._mask
+                if other is libmissing.NA:
+                    mask |= True
             else:
                 mask = self._mask | mask
 
-            with np.errstate(all="ignore"):
-                result = op(self._data, other)
+            if other is libmissing.NA:
```
Comment (jreback) [inline pandas/core/arrays/boolean.py:725]: can you add a comment or 2 about what is happening here

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/arrays/boolean/test_arithmetic.py:24]: i would call these left_array, right_array

File: pandas/tests/arrays/boolean/test_arithmetic.py [hunk 4]
```diff
@@ -13,30 +15,87 @@ def data():
...
+    tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "opname",
+    [
+        "floordiv",
+        "mod",
+        pytest.param(
+            "pow", marks=pytest.mark.xfail(reason="TODO follow int8 behaviour?")
```
Comment (jreback) [inline pandas/tests/arrays/boolean/test_arithmetic.py:69]: is there an issue number?

File: pandas/tests/arrays/boolean/test_arithmetic.py [hunk 5]
```diff
@@ -13,30 +15,87 @@ def data():
...
+    ],
+)
+def test_op_int8(a, b, opname):
+    op = getattr(operator, opname)
+    result = op(a, b)
+    expected = op(a.astype("Int8"), b.astype("Int8"))
+    tm.assert_extension_array_equal(result, expected)
+
+
+# Test generic charachteristics / errors
```
Comment (jreback) [inline pandas/tests/arrays/boolean/test_arithmetic.py:80]: sp?

File: pandas/tests/arrays/masked/test_arithmetic.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+    tm.assert_series_equal(result, expected)
+
+    # Series with Series
+    other = pd.Series([scalar] * len(data), dtype=data.dtype)
+    result = op(s, other)
+    expected = pd.Series(op(data, other.array))
+    tm.assert_series_equal(result, expected)
+
+
+# Test generic charachteristics / errors
```
Comment (jreback) [inline pandas/tests/arrays/masked/test_arithmetic.py:126]: sp?

**Review by jreback — CHANGES_REQUESTED** [review]
looks pretty good

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/boolean/test_arithmetic.py:24]: Renamed

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 4 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/boolean/test_arithmetic.py:69]: Not yet, will open one.

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 5 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/boolean/test_arithmetic.py:80]: Does "sp" point to the typo in this title? Or did you mean something else?

File: pandas/tests/arrays/boolean/test_arithmetic.py (hunk 4 above)
Comment (jorisvandenbossche) [reply pandas/tests/arrays/boolean/test_arithmetic.py:69]: -> https://github.com/pandas-dev/pandas/issues/34686

**Comment by jorisvandenbossche** [comment]
Updated for all comments

File: pandas/_testing.py [hunk 7]
```diff
@@ -2758,3 +2759,28 @@ def get_cython_table_params(ndframe, func_names_and_expected):
             if name == func_name
         ]
     return results
+
+
+def get_op_from_name(op_name: str) -> Callable:
```
Comment (jreback) [inline pandas/_testing.py:2764]: IIRC we have something similiar in internals, maybe we should consolidate (just a thought) and certainly. in a followon

**Review by jreback — APPROVED** [review]
lgtm.

**Comment by jreback** [comment]
thanks @jorisvandenbossche

**Pull request merged**